src/buildx/install.ts

```diff
@@ -231,3 +263,74 @@ export class Install {
     return releases[version];
   }
 }
+
+class InstallCache {
+  private readonly htcName: string;
```

Comment:
"hosted tool cache". This is the "poor-cache" method for self hosted runners. If binary is found in hosted tool cache use this one, otherwise fall back to GHA cache otherwise download or build the binary.